nixos/modules/services/misc/weechat.nix

```diff
@@ -0,0 +1,103 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.weechat;
+in
+
+{
+  options.services.weechat = {
+    enable = mkEnableOption "weechat";
+    init = mkOption {
+      description = "Weechat commands applied at start, one command per line.";
+      example = ''
+        /set relay.network.password correct-horse-battery-staple
+        /relay add weechat 9001
+      '';
+      type = types.str;
+      default = "";
+    };
+    root = mkOption {
+      description = "Weechat state directory.";
+      type = types.str;
+      default = "/var/lib/weechat";
+    };
+    scripts = mkOption {
+      description = "Custom scripts to be loaded by the weechat module.";
+      type = with types; listOf (submodule {
+        options = {
+          name = mkOption {
+            type = str;
+            description = "Path to the script in the derivation.";
```

Comment:
So `name` is actually a path?

Reply:
I was thinking about "name of the script", but well, there are only two hard things in computer science, right? ;)

But seriously, you're right, `path` is much better, however this can be dropped as you stated later on.